impl/src/context.rs

```diff
@@ -29,12 +33,25 @@ pub struct BuildableDependency {
   pub is_proc_macro: bool,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize)]
+#[derive(Debug, Clone, Eq, PartialOrd, Ord, Serialize)]
 pub struct DependencyAlias {
   pub target: String,
   pub alias: String,
 }
 
+// We only want equality for the first member as it can have multiple names pointing at it.
+impl Hash for DependencyAlias {
+  fn hash<H: Hasher>(&self, state: &mut H) {
+    self.target.hash(state);
+  }
+}
+
+impl PartialEq for DependencyAlias {
+  fn eq(&self, other: &Self) -> bool {
+    self.target == other.target
+  }
+}
```

Comment:
I've been bitten by some nasty bugs where hash or comparison functions don't cover the all of the data in the struct.

Is there another way to structure this, perhaps by using map from target string to alias (or &DependencyAlias)?